chain/chain/src/chain.rs

```diff
@@ -4530,6 +4530,36 @@ pub struct ApplyStatePartsResponse {
     pub sync_hash: CryptoHash,
 }
 
+#[derive(actix::Message)]
+#[rtype(result = "()")]
+pub struct LoadMemtrieRequest {
+    pub runtime_adapter: Arc<dyn RuntimeAdapter>,
+    pub shard_uid: ShardUId,
+    pub prev_state_root: StateRoot,
+    pub sync_hash: CryptoHash,
+}
+
+// Skip `runtime_adapter` and `epoch_manager`, because these are complex object that have complex logic
+// and many fields.
+impl Debug for LoadMemtrieRequest {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("LoadMemtrieRequest")
+            .field("runtime_adapter", &"<not shown>")
+            .field("shard_uid", &self.shard_uid)
+            .field("prev_state_root", &self.prev_state_root)
+            .field("sync_hash", &self.sync_hash)
+            .finish()
+    }
+}
+
+#[derive(actix::Message, Debug)]
+#[rtype(result = "()")]
+pub struct LoadMemtrieResponse {
+    pub load_result: Result<(), near_chain_primitives::error::Error>,
+    pub shard_id: ShardId,
+    pub sync_hash: CryptoHash,
```

Comment:
what does this hash represent/belongs to? it is not clear from the field name `sync_hash`, maybe add a comment about this?